src/compiler/checker.ts

```diff
@@ -16272,13 +16272,13 @@ namespace ts {
         // for finite but deeply expanding indexed accesses (eg, for `Q[P1][P2][P3][P4][P5]`).
         function isDeeplyNestedType(type: Type, stack: Type[], depth: number): boolean {
             // We track all object types that have an associated symbol (representing the origin of the type)
-            if (depth >= 5 && type.flags & TypeFlags.Object) {
+            if (depth >= 5 && type.flags & TypeFlags.Object && !isObjectOrArrayLiteralType(type)) {
                 const symbol = type.symbol;
                 if (symbol) {
                     let count = 0;
                     for (let i = 0; i < depth; i++) {
                         const t = stack[i];
-                        if (t.flags & TypeFlags.Object && t.symbol === symbol) {
+                        if (t.flags & TypeFlags.Object && t.symbol === symbol && !isObjectOrArrayLiteralType(t)) {

```

Comment:
If two types have the same symbol and one of them is already known not to be an array or object literal type, how could the other possibly be? Under what scenarios does this need to be checked here, too?

Reply:
Yeah, you're right.